### PR TITLE
Typeahead upgrade

### LINKED ISSUE #540: Errors and warnings in console for /definitions
There are a few errors and warnings in the /definitions page on normal load. We should strive for there to be no errors or warnings (no output at all really). Here is the dump I see. 2 warnings and an error
```
WARNING 1
react-ga.js:204 [react-ga] gaTrackingID is required in initialize()
warn @ react-ga.js:204
_initialize @ react-ga.js:385
initialize @ react-ga.js:429
./src/index.js @ index.js:10
__webpack_require__ @ bootstrap:782
fn @ bootstrap:150
0 @ withTracker.js:38
__webpack_require__ @ bootstrap:782
checkDeferredModules @ bootstrap:45
webpackJsonpCallback @ bootstrap:32
(anonymous) @ main.chunk.js:1


ERROR 1
index.js:1446 Warning: Failed prop type: You provided a `value` prop to a form field without an `onChange` handler. This will render a read-only field. If the field should be mutable use `defaultValue`. Otherwise, set either `onChange` or `readOnly`.
    in input (created by AutosizeInput)
    in div (created by AutosizeInput)
    in AutosizeInput (created by HintedInput)
    in div (created by HintedInput)
    in HintedInput (created by TypeaheadInput)
    in div (created by TypeaheadInput)
    in div (created by TypeaheadInput)
    in TypeaheadInput (created by WrappedInput)
    in WrappedInput (created by Typeahead)
    in div (created by Typeahead)
    in Typeahead (created by WrappedTypeahead)
    in WrappedTypeahead (created by WrappedTypeahead)
    in WrappedTypeahead (created by Typeahead)
    in Typeahead (created by OnClickOutside(Typeahead))
    in OnClickOutside(Typeahead) (created by Container)
    in Container (at FilterBar.js:41)
    in FilterBar (at SearchBar.js:11)
    in div (created by Col)
    in Col (at SearchBar.js:10)
    in div (created by Row)
    in Row (at SearchBar.js:9)
    in SearchBar (at PageDefinitions.js:162)
    in div (created by Grid)
    in Grid (at PageDefinitions.js:155)
    in PageDefinitions (created by Connect(PageDefinitions))
    in Connect(PageDefinitions) (at withTracker.js:33)
    in HOC (created by Route)
    in Route (at RehydrationProvider.js:61)
    in Switch (at RehydrationProvider.js:60)
    in main (at App.js:13)
    in div (at App.js:11)
    in App (at RehydrationProvider.js:59)
    in Router (created by BrowserRouter)
    in BrowserRouter (at RehydrationProvider.js:58)
    in Provider (at RehydrationProvider.js:57)
    in RehydrationDelayedProvider (at src/index.js:18)

WARNING 2
webpackHotDevClient.js:120 ./node_modules/antd/dist/antd.css (./node_modules/css-loader??ref--6-oneOf-3-1!./node_modules/postcss-loader/src??postcss!./node_modules/antd/dist/antd.css)
Warning

(168:11) Replace text-decoration-skip: ink to text-decoration-skip-ink: auto, because spec had been changed
```

### PR DESCRIPTION
**Pull request by storrisi**
As mentioned in #540 , i've upgraded Typeahead to the latest version.
From here, i've updated some required props all over the components.

I've also split the GitHub Selector to work on 2 separate fields, one for the namespace and one for the package.

Fixes #540

### CODE REVIEW

**Comment by storrisi** [comment]
I've found a bug on this new version while showing the Dropdown menu in a List.
To fix that behavior, i've implement a new custom Autocomplete component that wraps Typeahead and renders a custom Menu using Portals.

**Comment by storrisi** [comment]
I've updated the Autocomplete component to support Highlighting and custom children rendering as well. 
I've also updated e2e since the new version of Typeahead renders some DOM elements differently.
You can now review this implementation.

**Comment by storrisi** [comment]
I've updated E2E tests here using mocked returned data while making a contribution.

This should:
- avoid failure on E2E if we're exceeding the limits of GitHub
- avoid creating too much automated curations

**Comment by storrisi** [comment]
@dabutvin @jeffmcaffer are you able to take a look at this one? We need the new version of Typeahead in place to move forward on other topics that require this change.

File: src/components/GitHubSelector.js [hunk 1]
```diff
@@ -14,49 +14,100 @@ export default class GitHubSelector extends Component {
...
-        onSearch={this.getOptions}
-      />
+      <div className="horizontalBlock">
+        <AsyncTypeahead
+          id="github-namespace-selector"
+          className="selector-picker"
+          ref={component => (this._typeahead = component ? component.getInstance() : this._typeahead)}
+          useCache={false}
+          options={namespace.options}
+          placeholder={'Pick a namespace'}
```
Comment (dabutvin) [inline src/components/GitHubSelector.js]: This should be `User / Organization`

File: src/components/GitHubSelector.js [hunk 2]
```diff
@@ -14,49 +14,100 @@ export default class GitHubSelector extends Component {
...
+          onSearch={this.getOptions}
+        />
+
+        <AsyncTypeahead
+          id="github-component-selector"
+          className="selector-picker"
+          ref={typeahead => (this._componentselector = typeahead)}
+          useCache={false}
+          options={component.options}
+          placeholder={'Pick a component'}
```
Comment (dabutvin) [inline src/components/GitHubSelector.js]: This should be `Repo`

**Comment by dabutvin** [comment]
I've noticed that the tag picker is not refreshing when i change the repo.

File: src/actions/curationActions.js [hunk 3]
```diff
@@ -87,6 +87,7 @@ export function curateAction(token, spec) {
       },
       error => {
         dispatch(actions.error(error))
+        console.log(error)
```
Comment (dabutvin) [inline src/actions/curationActions.js]: remove console log.

Can we get this in the lint rule so it fails a build?

**Comment by storrisi** [comment]
Thanks @dabutvin i've fixed the behavior and the labels as well

**Comment by storrisi** [comment]
@dabutvin this is ready to be reviewed

**Comment by jeffmcaffer** [comment]
OK. looks good. Ship when ready

**Comment by dabutvin** [comment]
👍

**Pull request merged**